app/vmui/packages/vmui/src/pages/CustomPanel/QueryConfigurator/QueryConfigurator.tsx

```diff
@@ -113,6 +123,51 @@ const QueryConfigurator: FC<QueryConfiguratorProps> = ({
     handleToggleHideQuery(e, i);
   };
 
+  const { serverUrl } = useAppState();
+
+  const handlePrettifyQuery = async (el: HTMLButtonElement, i: number) => {
```

Comment:
It would be beneficial to move this function to hooks. For reference, you can look at [useRelabelDebug](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/5aed369132b3e38d85bb29709b23d861503d9ff0/app/vmui/packages/vmui/src/pages/Relabel/hooks/useRelabelDebug.ts)

Reply:
Thanks, I tried `const { getPrettifiedQuery, prettifyError } = usePrettifyQuery();`,
(with prettifyError as an array, because it's separate for each query)).
It mostly works (see [ea3d0bd](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4694/commits/ea3d0bd307631e099632fe5dbce82c261e32c12f)), although there's a problem that prettifyErrors don't clean up queryErrors (and vice versa).

I'm going to reuse queryErrors...